src/HomePage/Header/Header.js

```diff
@@ -0,0 +1,141 @@
+import { Avatar } from "forkui-lib";
+import { useThemeContext } from "../../AllContext/themeContext";
+import { calculateLengthOfCart } from "../../Logic/CalcLengthOfCart";
+import { SetLanguageModel } from "./SetLanguageModel/SetLanguageModel";
+import { Button } from "forkui-lib";
+import { useLoginContext } from "../../AllContext/LoginContext";
+import { AiFillFire } from "react-icons/ai";
+export const Header = () => {
+  const { login, setLogin } = useLoginContext();
+  const [showLangModel, setShowLangModel] = useState(false);
+  const { language, setLanguage } = useLangContext();
+  const { theme, setTheme } = useThemeContext();
+  const [displaySidebar, setDisplaySidebar] = useState(false);
+  const { state } = useMyReducer();
+  const toggleModeHandler = () => {
+    theme === "light" ? setTheme("dark") : setTheme("light");
+  };
+  return (
+    <div className="header">
+      <div
+        className="sidebar"
+        style={{ display: displaySidebar ? "block" : "none" }}
+      >
+        <div className="sidebar__profile">
+          <span
+            className="sidebar__profile__cancel"
+            onClick={() => setDisplaySidebar(false)}
+          >
+            <FaTimes size={20} color="white" />
+          </span>
+          <span className="sidebar__profile__avatar">
+            <Link to="/profile">
+              <Avatar size="lg" src="https://bit.ly/3cQDipG" />
+            </Link>
+          </span>
+          <span className="sidebar__profile__profile-name">Rashita</span>
+        </div>
+        <div className="sidebar__nav-links">
+          <span
+            onClick={() => {
+              setDisplaySidebar(false);
+            }}
+          >
+            <Link to="/">{language.home}</Link>
+          </span>
+          <span
+            onClick={() => {
+              setDisplaySidebar(false);
+            }}
+          >
+            <Link to="/products">{language.products}</Link>
+          </span>
+          <span
+            onClick={() => {
+              setDisplaySidebar(false);
+            }}
+          >
+            <Link to="/cart">{language.cart}</Link>
+          </span>
+          <span
+            onClick={() => {
+              setDisplaySidebar(false);
+            }}
+          >
+            <Link to="/wishlist">{language.wishlist}</Link>
+          </span>
+        </div>
+      </div>
+      <div className="bars-brand">
+        <span
+          className="sidebar__bars"
+          onClick={() => setDisplaySidebar(!displaySidebar)}
+        >
+          {<FaBars size={25} color="#374151" />}
+        </span>
+        <span className="heading-icon-text">
+          <span className="proart-icon">
+            <AiFillFire size={25} color="#20a546" />
+          </span>
+          <Link to="/">{language.proArt}</Link>
+        </span>
+      </div>
+      <div className="nav-icons">
+        <span>{<AiOutlineSearch size={25} color="#374151" />}</span>
```

Comment:
I observed that you have hardcoded color values & passed those. It would be difficult to manage changes down the line.  It would help to have the colors stored in variables declared at one place in a file & then those variables passed. For example:
`const PRIMARY_COLOR: "374151";`

& then in your component pass the color using variable `color=PRIMARY_COLOR`

Reply:
Yes! Done that